keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1261 @@
+handling of all existing Pods without attached CCI drivers by the existing
+CPU manager and the triggering of Pod allocation done via the driver.
+The policy also takes care for fallback options, proper cluster start and 
+plugin failure handling.
+
+In the alpha version of the Resource Manager applies plugins to manage resource
+allocations for Pods which are explicitly marked to be handled by an allocation 
+plugin.  The overall architecture consists of a manager component – the Resource
+Manager inside the container manager, an optional Pod spec argument to identify which 
+Pods to be processed by the manager, a compute resource store – a component used to sync
+the cpuset state of the cpu manager and resource plugins and  a new policy inside cpu
+manager which can query the compute resource store for any allocations and allocation-removals
+done by the plugin.
+
+![image](ExternallyManagedResourceExtension.jpg)<br>
+Figure 1: CCI Resource Manager Architecture inside Kubelet
+
+    Pod Spec “CCIDriverName Extension”
+    type PodSpec {
+       …
+    	 // CCIResourceDriver is an optional parameter which can identify if a Pod resource manager (cpu an memory)
+	    // has to be handled by a pluggable CCI resource driver 
+	    // This is an alpha field and requires enabling the
+    	 // CCIResourceManager feature gate.
+	    //
+    	 // This field is immutable.
+	    //
+	    // +featureGate=CCIResourceManager
+	    // +optional
+	    CCIResourceDriver string `js”n:”cciResourceDriver,omitem”ty” protob”f:”bytes,40,opt,name=cciResourceDri”er”`
+     …}
+
+The proposed extension relies on a new optional argument inside the Pod spec to `driverName`. 
+We use this optional argument to drive the allocation process. If the driver name is not provided,
+the Pod resources will be allocated through the standard CPU manager within t ubeletlet. An
+alternative association approach can be considered during the implementation which can avoid
+Pod spec changes.
+
+1. CCI Resource Manager<br>
+These changes will need a separate resource manager component inside the container 
+manager which will handle plugin management tasks, including plugin registration and 
+invocations of registered plugins.  These tasks will be managed via grpc for Pods who 
+require one or more resource management plugin(s) to get successfully allocated.  In 
+the alpha version we provide to entry points to plugins on admission,  addContainer 
+and removeContainer lifetime events (plugins have to implement Admit, AddResource 
+and RemoveResource handlers accordingly). 
+
+2. CCI Resource Store<br>
+The Resource Manager uses a store to keep track of resourcesets which can include 
+cpusets, memory, etc ..  allocated by CCI Drivers. This information is gathered 
+per container. The store will be used by a new cpu manager policy to synchronize 
+the state between cpu manager and pluggable CCI Drivers.  The store offers the 
+following interface to manage resource sets.
```

Comment:
This will apply also to memory and device managers in beta and after beta, right?
Echoing a previous comment of mine, state synchronziation makes me pause. I'd really like to learn which other options do we have here and why they were discarded

Reply:
Currently pausing on state synchronization with limiting options, per Kevin's suggestion.